Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedPlatformLayer.cpp

```diff
@@ -932,6 +932,14 @@ void CoordinatedPlatformLayer::flushCompositingState()
     else
         layer.setContentsLayer(nullptr);
 
+#if ENABLE(DAMAGE_TRACKING)
+    if (m_pendingChanges.contains(Change::Damage) && layer.hasContentsLayer()) {
+        // Layers with content layer should have no explicit damage other than the one synthetized due to transform changes.
+        // FIXME: Remove special handling of content layers.
+        layer.setDamage({ });
+    }
+#endif
```

Comment:
Can we handle this before when we set the damage? The contents layer hasn't been set yet, but we know if the layer will have it 

```
if (m_committedContentsBuffer || (m_imageBackingStore && m_imageBackingStoreVisible))
```